pennylane/optimize/shot_adaptive.py

```diff
@@ -479,7 +478,8 @@ def step(self, objective_fn, *args, **kwargs):
 
             argmax_gamma = np.unravel_index(np.argmax(gamma), gamma.shape)
             smax = max(s[argmax_gamma], 2)
-            self.s[idx] = np.squeeze(np.int64(np.clip(s, min(2, self.min_shots), smax)))
+
+            self.s[idx] = np.int64(np.clip(s, min(2, self.min_shots), smax))

```

Comment:
did you encounter anything where this squeezing might have been useful, or have some insight as to why it might have been there before? i'm fine to drop it, just curious in case we might be breaking some un-tested corner case just to fix another.

Reply:
My intuition is that it was added with the assumption that users will not create trainable parameters with unnecessary extra dimensions, and perhaps also to mitigate any extra dimensions that are added during the matrix arithmetic.

As far as matrix arithmetic is concerned though, only operators that update the matrix element-wise are present, so the matrix dimensions wouldn't change, making the need for squeezing moot.